docs/proposals/custom-sidecar-config.md

```diff
@@ -0,0 +1,172 @@
+# Custom Container Configuration
+
+## Introduction
+Users and field team have requested access to custom configuration of kubernetes sidecar and init containers.
+See [#3925](https://github.com/kumahq/kuma/issues/3925).
+This intuitively aligns with the ProxyTemplate configuration,
+which provides a free-form configuration of Envoy.
+The current plan is to introduce a new CRD,
+similar to ProxyTemplate,
+which can modify sidecar config utilizing [jsonpatch](https://jsonpatch.com).
+
+## Custom Resource Definition
+The new CRD will be named ContainerPatch.
+It will allow for customer configuration of
+both sidecar and init containers.
+It will be namespace scoped.
+The webhook will validate that it can only be applied in a namespace where Kuma CP is running.
+The spec will contain an array of jsonpatch strings which describe the modifications to be performed.
+There is no Universal mode equivalent.
+
+### Example
+
+```yaml
+apiVersion: kuma.io/v1alpha1
+kind: ContainerPatch
+metadata:
+  name: container-patch-1
+spec:
+  sidecarPatch:
+    - op: add
+      path: /securityContext/privileged
+      value: true
+  initPatch:
+    - op: add
+      path: /securityContext/runAsNonRoot
+      value: true
+    - op: remove
+      path: /securityContext/runAsUser
+
+```
+
+This will change the `securityContext` section of `kuma-sidecar` container from
+
+```yaml
+      securityContext:
+        runAsGroup: 5678
+        runAsUser: 5678
+```
+
+to
+
+
+```yaml
+      securityContext:
+        runAsGroup: 5678
+        runAsUser: 5678
+        priviledged: true
+```
+
+and similarly change the `securityContext` section of the `init` container from
+
+```yaml
+      securityContext:
+        capabilities:
+          add:
+          - NET_ADMIN
+          - NET_RAW
+        runAsGroup: 0
+        runAsUser: 0
+```
+
+to
+
+```yaml
+      securityContext:
+        capabilities:
+          add:
+          - NET_ADMIN
+          - NET_RAW
+        runAsGroup: 0
+        runAsNonRoot: true
+```
+
+## Workload Matching
+A `ContainerPatch` will be matched to a Pod via an annotation on the workload.
+Each annotation may be an ordered list of `ContainerPatch` names,
+which will be applied in the order specified.
+
+We will include a configuration option to Kuma CP
+to specify a default list of `ContainerPatch` patches to apply.
```

Comment:
I think you are missing mentioning adding a default in the CP conf. Would be nice to have this explicitly say and confirm the name, placement and default of this CP config.